ui-admin/src/util/withStateResetOnEnvChange.tsx

```diff
@@ -0,0 +1,20 @@
+import React, { ComponentType, useEffect, useState } from 'react'
+import { StudyEnvContextT } from 'study/StudyEnvironmentRouter'
+
+export const withStateResetOnEnvChange = <T, >(WrappedComponent: ComponentType<T>) => {
+  const WithStateResetOnEnvChange = (props: T & { studyEnvContext: StudyEnvContextT }) => {
+    const { studyEnvContext } = props
+    const [key, setKey] = useState(studyEnvContext.currentEnvPath)
+
+    useEffect(() => {
```

Comment:
I don't understand why we need this hook -- if studyEnvContext.currentEnvPath is what we're listening for for changes, why can't we use that as the `key` directly?

Reply:
ah you're totally right- this was some leftover from an earlier iteration. just cleaned this up